### PR TITLE
ulogd: add extra space for null char for strings

### PR DESCRIPTION
**Pull request by commodo**
Maintainer: @psycho-nico 
Compile tested: https://github.com/lede-project/source/commit/7d70ad66ac7e30e7387debbedd6ec6c996ce3257  LEDE 17.01 (mpc85xx)
Run tested: https://github.com/lede-project/source/commit/7d70ad66ac7e30e7387debbedd6ec6c996ce3257  LEDE 17.01 (mpc85xx)
Description:

Fixes  https://github.com/openwrt/packages/issues/4090

Signed-off-by: Alexandru Ardelean <ardeleanalex@gmail.com>

### CODE REVIEW

**Comment by commodo** [comment]
validated also by @LEDEuser

@hnyman 
if maintainer times out on this, i'm fine to take this package

**Comment by hauke** [comment]
In all the places I checked the buffers you changed were used with strncpy() which should not overflow. When the string is bigger it will be truncated and *no* null byte added, this could cause a problem, but is not fixed with your patch.  I think this is probably only hiding some other problem.

**Comment by commodo** [comment]
@hauke 
i'll admit this patch is a bit half-assed 
it was i implemented internally for resolving the issue [when we updated the base system from an old openwrt to a newer lede] ; for mpc85xx it seems to still work

the problem first starts with the inconsistent use of `ULOGD_MAX_KEYLEN`
where in some places it's  `char name[ULOGD_MAX_KEYLEN+1];` and in some places [where i patched] it's `char name[ULOGD_MAX_KEYLEN];`

but the fail as i recall happened [in our case] in  `src/ulogd.c` around line 671, where instead of using `strncpy()` [or any variant for strcpy()] it's `memcpy(pi->id, pi_id, sizeof(pi->id));`
id is defined as `char id[ULOGD_MAX_KEYLEN+1];` and `char pi_id[ULOGD_MAX_KEYLEN];`

basically in this case, `memcpy()` is segfaulting by copying from the stack 1 random byte

now, this particular case could be just resolved simply by replacing that `memcpy()` with `strncpy()` or better yet `strlcpy()` [as this one guarantees a null char]

but also, i'd say `ulogd` would need a bit of refactor to change all `char name[ULOGD_MAX_KEYLEN+1];` to  `char name[ULOGD_MAX_KEYLEN];`, define `ULOGD_MAX_KEYLEN` as 32 [not 31 as it is now]

i could also make some time to send some patches to `ulogd` to properly address this issue
i already sent this one as is ; no reply yet [ or i'm not doing something correctly ]

**Comment by commodo** [comment]
@hauke @hnyman @LEDEuser 
closing this for now ;

i have an open discussion with the upstream discussion with the maintainer of ulogd [ Eric Leblond ]
he counter-proposed another patch, which i'm testing

will come back later with a conclusion

thanks
sorry for the noise
